### PR TITLE
Tli26

### PR DESCRIPTION
**Pull request by caroline666-t**
2 main features:
1. Limit daily modifications of one item to 3 
2. Limit items with same due date to 8

### CODE REVIEW

**Comment by Emmathy108** [comment]
> 2 main features:
> 
> 1. Limit daily modifications of one item to 3
> 2. Limit items with same due date to 8

**Comment by PohotonicCrystal** [comment]
A clear PR message

File: design/core.png
Comment (Emmathy108) [inline design/core.png:1]: Updating UML diagrams promptly is good~

File: TodoItems.Core/ITodosRepository.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace TodoItems.Core
+{
+    public interface ITodosRepository
+    {
+        List<TodoItem> FindAllTodoItemsHaveTheSameDueDate(DateTime dueDate);
```
Comment (PohotonicCrystal) [inline TodoItems.Core/ITodosRepository.cs:5]: retuen a number maybe better

File: TodoItems.Test/TodoItemServiceTest.cs
Comment (Emmathy108) [inline TodoItems.Test/TodoItemServiceTest.cs:1]: typo: ninth is correct, not nineth

File: TodoItems.Core/TodoItem.cs [hunk 2]
```diff
@@ -1,9 +1,37 @@
-﻿namespace TodoItems.Core;
+﻿using System;
+
+namespace TodoItems.Core;
 
 public class TodoItem
 {
-    public string GetId()
+    public string Id { get; init; }
```
Comment (PohotonicCrystal) [inline TodoItems.Core/TodoItem.cs:7]: need a constructor

File: TodoItems.Core/TodoItem.cs [hunk 3]
```diff
@@ -1,9 +1,37 @@
...
+
+namespace TodoItems.Core;
 
 public class TodoItem
 {
-    public string GetId()
+    public string Id { get; init; }
+    public string Description { get; set; }
+    public DateTime CreatedTime { get; set; } = DateTime.Now;
+    public IList<Modification> Modifications { get; set; }
```
Comment (anyan-0110) [inline TodoItems.Core/TodoItem.cs:10]: recommend Modifications to be private

File: TodoItems.Test/TodoItemTest.cs [hunk 4]
```diff
@@ -5,9 +5,53 @@ namespace TodoItems.Test;
...
     [Fact]
-    public void should_return_2_when_add_1_1()
+    public void Should_throw_exception_when_fourth_modification()
     {
-        var todoItem = new TodoItem();
-        Assert.Equal("1", todoItem.GetId());
+        DateTime today = DateTime.Today;
+        List<Modification> threeTodayModifications =
+        [
+            new Modification(today.AddHours(9)),
```
Comment (PohotonicCrystal) [inline TodoItems.Test/TodoItemTest.cs:13]: Call modify 3 times is more graceful

File: TodoItems.Core/ITodosRepository.cs (hunk 1 above)
Comment (caroline666-t) [reply TodoItems.Core/ITodosRepository.cs:5]: You are right, PohotonicCrystal

**Pull request closed without merging**